### DISCUSSION TITLE
Migrating the ecosystem

### DISCUSSION THREAD

**Discussion by phpnode** [peggyjs/peggy#28] [Ideas]
When v1 is safely out of the door and we're happy with it we should start reaching out to the broader ecosystem to encourage people to switch over. There's a bunch of pegjs plugins, prettier plugins and syntax highlighting rules that are in active use and it'd be great if we could support them.

At some point we will also need to introduce our own file extension, but `.pegjs` should continue to work for the foreseeable future.

We should also write an announcement that explains the reasons for the fork (in a tactful way), and write some simple instructions to make switching over easier. It may be possible to get npm to deprecate the pegjs package if David can talk to them, but I don't know whether that's desirable or whether he'd be comfortable doing so - it would help with adoption though.

It would also be great if we can get some of the more prolific / well known users of PEG.js to support this effort. I know @mikeal uses it on some of his projects, there are likely many more.

We should also look at the dependency graph on npm and github and reach out to the bigger projects that consume it encouraging them also to make the switch.

**Comment by hildjj** [comment]
See https://github.com/pegjs/pegjs/issues/667

**Comment by hildjj** [comment]
https://twitter.com/hildjj/status/1383169196981317635

**Comment by hildjj** [comment]
See https://github.com/SrTobi/code-pegjs-language/pull/5

**Reply by hildjj** [reply]
According to the above conversation, I just published a repack of @SrTobi 's code at https://marketplace.visualstudio.com/items?itemName=PeggyJS.peggy-language

I'd love it if someone would try it out and see if it works for them.

**Reply by SrTobi** [reply]
@hildjj I tried it and it's pretty cool! The new features are also pretty great! I found a little bug and made two pull requests. Great work!

**Comment by pjmolina** [comment]
Hi. I just migrated the plugin ts-pegjs to use peggy. 
https://github.com/metadevpro/ts-pegjs

Regards,

**Reply by hildjj** [reply]
Make sure you're tracking #117 for the 1.2 release, please.  

There is a new opcode `PUSH_EMPTY_STRING`, and we need your input on whether `MATCH_CHAR_CLASS` and `MATCH_REGEXP` should be the same opcode, or whether `MATCH_CHAR_CLASS` should get a new number.

We'll make sure to keep you in the loop for anything that might affect your code.

**Reply by hildjj** [reply]
Thanks for switching!

**Reply by pjmolina** [reply]
Thanks @hildjj  for keeping the fire alive. 
I will take a look to #117 and will come back with my feedback.

**Comment by SrTobi** [comment]
Hi guys, great work with Peggy. I added a message (with dramatic caption :smile:) to the readme of my [PEG.js extension](https://marketplace.visualstudio.com/items?itemName=SirTobi.pegjs-language) and suggest to switch to the peggy extension.

**Comment by MarcelBolten** [comment]
Hi,
I started to migrate the "old" plugin [phpegjs](https://github.com/nylen/phpegjs) because I want to use it with Peggy.
Please have a look at [PHPeggy](https://github.com/MarcelBolten/phpeggy). So far it is only compatible with Peggy@1.0.0.
Cheers

**Comment by hildjj** [comment]
See https://github.com/peggyjs/peggy/issues/193
